### PR TITLE
Take flush_time into account for scheduling background flush of the Buffer

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow-up for: #22986

### CODE REVIEW

**Comment by azat** [comment]
Note, that this is required for #35072 (for the test)

File: src/Storages/StorageBuffer.cpp [hunk 1]
```diff
@@ -1000,7 +1000,8 @@ void StorageBuffer::reschedule()
 
     size_t min = std::max<ssize_t>(min_thresholds.time - time_passed, 1);
     size_t max = std::max<ssize_t>(max_thresholds.time - time_passed, 1);
-    flush_handle->scheduleAfter(std::min(min, max) * 1000);
+    size_t flush = std::max<ssize_t>(flush_thresholds.time - time_passed, 1);
+    flush_handle->scheduleAfter(std::min(std::min(min, flush), max) * 1000);
```
Comment (antonio2368) [inline src/Storages/StorageBuffer.cpp:1004]:
```suggestion
    flush_handle->scheduleAfter(std::min({min, flush, max}) * 1000);
```

**Review by antonio2368 — APPROVED** [review]
LGTM with a small C++ comment which is not necessary.

File: src/Storages/StorageBuffer.cpp (hunk 1 above)
Comment (azat) [reply src/Storages/StorageBuffer.cpp]: Nice catch, thanks! Applied & rebased.

**Pull request merged**